prime-router/src/test/kotlin/fhirengine/azure/FHIRRouterIntegrationTests.kt

```diff
@@ -0,0 +1,1066 @@
+                    emptyMap(),
+                    emptyList()
+                ),
+                txn
+            )
+        }
+
+        return report
+    }
+
+    private fun createReportsWithLineage(
+        reportContents: String,
+        topic: Topic = Topic.FULL_ELR,
+    ): Pair<Report, Report> {
+        val receivedBlobUrl = BlobAccess.uploadBlob(
+            "receive/mr_fhir_face.fhir",
+            reportContents.toByteArray(),
+            getBlobContainerMetadata()
+        )
+
+        val convertedBlobUrl = BlobAccess.uploadBlob(
+            "convert/mr_fhir_face.fhir",
+            reportContents.toByteArray(),
+            getBlobContainerMetadata()
+        )
+
+        val convertReport = createReport(
+            Report.Format.FHIR,
+            TaskAction.convert,
+            TaskAction.route,
+            Event.EventAction.ROUTE,
+            topic,
+            0,
+            null,
+            convertedBlobUrl
+        )
+
+        val receiveReport = createReport(
+            Report.Format.FHIR,
+            TaskAction.receive,
+            TaskAction.convert,
+            Event.EventAction.CONVERT,
+            topic,
+            0,
+            convertReport,
+            receivedBlobUrl
+        )
+
+        return Pair(receiveReport, convertReport)
+    }
+
+    private fun checkActionTable(expectedTaskActions: List<TaskAction>) {
+        ReportStreamTestDatabaseContainer.testDatabaseAccess.transact { txn ->
+            val actionRecords = DSL.using(txn)
+                                   .select(Tables.ACTION.asterisk())
+                                   .from(Tables.ACTION)
+                                   .fetchInto(
+                                       Action::class.java
+                                   )
+
+            for (i in 0 until actionRecords.size) {
+                assertEquals(expectedTaskActions.get(i), actionRecords.get(i).actionName)
+            }
+        }
+    }
+
+    private data class ActionLogRecordContent (
+        val index: Int,
+        val reportId: UUID?,
+        val type: ActionLogLevel?,
+        val scope: ActionLogScope?
+    )
+
+    private val expectedButStrangeObservationFilterEntry = listOf(
+        ActionLogRecordContent(
+            0,
+            reportId = null,
+            scope=ActionLogScope.report,
+            type=ActionLogLevel.info
+        )
+    )
+
+    private fun checkActionLogTable(expectedContentList: List<ActionLogRecordContent>) {
+        ReportStreamTestDatabaseContainer.testDatabaseAccess.transact { txn ->
+            val actionLogRecords = DSL.using(txn)
+                                      .select(Tables.ACTION_LOG.asterisk())
+                                      .from(Tables.ACTION_LOG)
+                                      .fetchInto(ActionLog::class.java)
+
+            assert(actionLogRecords.size >= expectedContentList.size)
+
+            for (expectedActionLogRecordContent in expectedContentList) {
```

Comment:
Can we enhance these assertions more to look at the actual message that is generated?  That will be what is actually returned from the submission endpoint and so we want to make sure that it's correct.